distrib_rl/Agents/MARLAgent.py

```diff
@@ -90,6 +92,8 @@ def gather_timesteps(self, policy, env, num_timesteps=None, num_seconds=None, nu
         for i in range(agents_to_save):
             trajectories[i].final_obs = next_obs[i]
             experience_trajectories.append(trajectories[i])
+            if trajectory_callback:
+                trajectory_callback(trajectories[i])
 
         return experience_trajectories
```

Comment:
does this still need to return the experience data here? if not, the callback should not be an optional argument